### PR TITLE
Move spec and integrate other changes (format_version: 0.3.2)

### PR DESCRIPTION
**Pull request by FynnBe**
Move over spec definition, validation and generated documentation from https://github.com/bioimage-io/python-bioimage-io

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,213 +1,49 @@
-## Model Description File Specification
+# Model Description File Specification
 
-This specification defines a file format for representing pretrained AI models in [YAML format](https://en.wikipedia.org/wiki/YAML). This format is used to describe models hosted in the [BioImage.IO](https://bioimage.io) model repository site.
-
-To get a quick overview of the config file, see an example file [here](https://github.com/bioimage-io/pytorch-bioimage-io/blob/master/specs/models/unet2d_nuclei_broad/UNet2DNucleiBroad.model.yaml).
+[This specification](./generated/bioimageio_model_spec.md) defines a file format for representing pretrained AI models in [YAML format](https://en.wikipedia.org/wiki/YAML). This format is used to describe models hosted on the [BioImage.IO](https://bioimage.io) model repository site.
```
Comment (constantinpape) [inline README.md:3]: I think eventually we should also render this on the bioimage.io doc page and then give that as the main reference.
But we can like to the md on github for now.

**Review by constantinpape — COMMENTED** [review]
I am very much in favor of moving this, but we should maybe first add the changes in #79 and integrate them into the doc generated by the spec, so that doesn't get lost.

**Comment by FynnBe** [comment]
yes, let's add the ideas of #70, #81, #82 and #83 as 0.3.2 here and not merge them individually (and then loose their changes in  merge conflicts of this PR)...

**Comment by oeway** [comment]
Maybe we should add a changelog section in the README or add a standalone CHANGELOG file to keep track of the changes of different version.

**Comment by FynnBe** [comment]
> Maybe we should add a changelog section in the README or add a standalone CHANGELOG file to keep track of the changes of different version.

7a204ac adds it to the readme

**Comment by FynnBe** [comment]
tests only fail due to problems in https://github.com/deepimagej/models
Maybe we should make this repository independent of others in the future.

**Comment by constantinpape** [comment]
> Maybe we should make this repository independent of others in the future.

Yes, that's a good idea; we should add an example model containing each weight format here then for the tests.

**Pull request merged**